### PR TITLE
scripts/line_length: Use temporary config file and only operate on staged files

### PR DESCRIPTION
**Pull request by Holzhaus**

### CODE REVIEW

File: scripts/line_length.py [hunk 1]
```diff
@@ -17,48 +20,50 @@ def main(argv: Optional[Sequence[str]] = None) -> int:
...
-                    "DerivePointerAlignment: false, "
-                    "ReflowComments: false, "
-                    "SpaceAfterTemplateKeyword: false, "
-                    "SpacesBeforeTrailingComments: 1}",
-                    f"{filename}",
+
+                fd.seek(0)
+                cmd = [
+                    "clang-format",
+                    "--style=file",
```
Comment (daschuer) [inline scripts/line_length.py]: Why do you don't keep the -i option?

File: scripts/line_length.py [hunk 2]
```diff
@@ -17,48 +20,50 @@ def main(argv: Optional[Sequence[str]] = None) -> int:
...
+    proc = subprocess.run(
+        ["clang-format", "--dump-config"], capture_output=True, text=True
+    )
+    proc.check_returncode()
+
+    with tempfile.TemporaryDirectory(prefix="clang-format") as tempdir:
+        # Create temporary config with ColumnLimit enabled
+        configfile = os.path.join(tempdir, "clang-format-style")
+        with open(configfile, mode="w") as configfp:
+            configfp.write(
```
Comment (daschuer) [inline scripts/line_length.py]: How about joining the lines with a comma and pass all this as a -style={}
paramerer. This would avoid using a temp file.

**Comment by daschuer** [comment]
@Holzhaus see my comments.

**Comment by daschuer** [comment]
Unfortunately now the two clang-format calls are using different rules. 
I think we need to call the dump run with a cpp file that it is able tow find the sibiling .clang-format file

File: scripts/line_length.py (hunk 2 above)
Comment (Holzhaus) [reply scripts/line_length.py]: I don't think that's feasible. The format looks a bit different and has section markers like `---` and also has yaml-like indented lists which won't work on a single line. I don't think it's worth the hassle. On Linux, the tempfile is created in `/tmp` which is usually a `tmpfs` inside the RAM anyway.

File: scripts/line_length.py (hunk 1 above)
Comment (Holzhaus) [reply scripts/line_length.py]: In order to specify a custom tempfile, I had to pass the file content via stdin (hence `-i` makes no sense here)

**Comment by Holzhaus** [comment]
Please retest. This now uses `src/mixxx.cpp` for getting the config. I also added git support so that this only changes changed lines, not the whole file.

**Comment by daschuer** [comment]
Works perfectly now. Thank you.

**Pull request merged**